src/basic/chase.c

```diff
@@ -784,6 +784,18 @@ int chase_and_unlink(
         return 0;
 }
 
+int chase_and_pin(const char *path, const char *root, ChaseFlags chase_flags, char **ret_filename) {
+        int pfd, r;
+
+        assert(!(chase_flags & (CHASE_NONEXISTENT|CHASE_STEP|CHASE_PARENT|CHASE_EXTRACT_FILENAME)));
```

Comment:
In the below, CHASE_PARENT and CHASE_EXTRACT_FILENAME are added. Hence, it is not necessary to set it by the caller, but I do not think it is not necessary to trigger the assertion...
So, please drop the two flags from the assertion.